signer/src/transaction_signer.rs

```diff
@@ -160,6 +168,35 @@ struct AcceptedSigHash {
     public_key: PublicKeyXOnly,
 }
 
+/// An enum identifying requests for which we can sign for on stacks only once
+/// per tenure.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub enum StacksSignRequestId {
```

Comment:
Can we implement display here too, so that our logs aren't too painful on our precious eyes.